### PR TITLE
Typescript export should acknowledge enum #29

### PR DESCRIPTION
**Pull request by frimpsss**
(Edit: @blackmann -> addresses #29)

### CODE REVIEW

File: base/src/lib/export-schema.ts [hunk 1]
```diff
@@ -213,7 +213,17 @@ async function exportToTypescript(
           break
         }
       }
+      case "string": {
+        const containsEnums: boolean = definition?.enum
+        if (containsEnums) {
+          const extractedEnums = definition?.enum.map(item => (item))
```
Comment (blackmann) [inline base/src/lib/export-schema.ts]: Any reason you're mapping the string back to a string again?

**Review by blackmann — COMMENTED** [review]
Thanks for this @frimpsss 

Also can you include tests for this?

File: base/src/lib/export-schema.ts [hunk 2]
```diff
@@ -213,7 +213,17 @@ async function exportToTypescript(
           break
         }
       }
+      case "string": {
+        const containsEnums: boolean = definition?.enum
```
Comment (blackmann) [inline base/src/lib/export-schema.ts]: `definition` is not optional. No need for `?`
```suggestion
        const containsEnums: boolean = definition.enum
```

Comment (blackmann) [reply base/src/lib/export-schema.ts]: To reduce code, maybe just put this check inside the parenthesis of the if

File: base/src/lib/export-schema.ts (hunk 1 above)
Comment (frimpsss) [reply base/src/lib/export-schema.ts]: I thinks its not necessary. ill remove it

File: base/src/lib/export-schema.ts (hunk 2 above)
Comment (frimpsss) [reply base/src/lib/export-schema.ts]: okay

File: base/src/lib/__snapshots__/export-schema.test.ts.snap [hunk 3]
```diff
@@ -53,3 +53,35 @@ exports[`export-schema > typescript > should return typescript definition [inlin
...
+  friends_contacts: {
+    email: string
+    email: string
+    name: string
+    name: string
+  }[]
+  name: string
+  name: string
+  tags?: string[]
+  gender?: male | female
```
Comment (blackmann) [inline base/src/lib/__snapshots__/export-schema.test.ts.snap]: This should be

```suggestion
  gender?: 'male' | 'female'
```

Comment (blackmann) [reply base/src/lib/__snapshots__/export-schema.test.ts.snap]: I would assume adding the `gender` definition will affect the other test snapshots but it isn't. I don't know if those changes were intentionally removed.

But you won't need to write a whole new test. Just adding the schema definition for an enum is fine. The other tests will capture it.

If I'm unclear, text me in the Discord

**Pull request merged**